File: CHL-Frontend/src/app/model/medical-data.ts
export class MedicalData {
    constructor(
        public id?:string|number|null,
        public patientId?: string|number|null,
        public isCurrent?: string|number|null,
        public height?: string|number|null,
        public weight?: string|number|null,
        public bmi?: string|number|null,
        public pulseRate?: string|number|null,
        public bloodPressure?: string|number|null,
        public bloodGroup?: string|number|null,
        public allergies?: string|number|null,
        public immunizations?: string|number|null,
        public chronicHealthProblems?: string|number|null,
        public familyMedicalHistory?: string|number|null,
        public uploadDate?: string|number|null
    ) { }
}

File: CHL-Backend/src/main/java/com/CentralHealthLog/entity/MedicalData.java
package com.CentralHealthLog.entity;

import com.fasterxml.jackson.annotation.JsonFormat;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.*;
import java.util.Date;

@Entity
@Data
@AllArgsConstructor
@NoArgsConstructor
public class MedicalData {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private Long patientId;
    private Boolean isCurrent;
    private Double height;
    private Double weight;
    private Double bmi;
    private Double pulseRate;
    private String bloodPressure;
    private String bloodGroup;
    private String allergies;
    private String immunizations;
    private String chronicHealthProblems;
    private String familyMedicalHistory;

    @JsonFormat(pattern="yyyy-MM-dd")
    private Date uploadDate;

}

